src/lib/parse-cdn.ts

```diff
@@ -1,31 +1,51 @@
 import { embeddedWriteKey } from './embedded-write-key'
-const regex = /(https:\/\/.*)\/analytics\.js\/v1\/(?:.*?)\/(?:platform|analytics.*)?/
-
-export function getCDN(): string {
-  let cdn: string | undefined = undefined
 
+const analyticsScriptRegex = /(https:\/\/.*)\/analytics\.js\/v1\/(?:.*?)\/(?:platform|analytics.*)?/
+const getCDNUrlFromScriptTag = (): string | undefined => {
+  let cdn: string | undefined
   const scripts = Array.prototype.slice.call(
     document.querySelectorAll('script')
   )
-
   scripts.forEach((s) => {
     const src = s.getAttribute('src') ?? ''
-    const result = regex.exec(src)
+    const result = analyticsScriptRegex.exec(src)
 
     if (result && result[1]) {
       cdn = result[1]
     }
   })
+  return cdn
+}
 
-  // it's possible that the CDN is not found in the page because:
-  // - the script is loaded through a proxy
-  // - the script is removed after execution
-  // in this case, we fall back to the default Segment CDN
-  if (!cdn) {
-    return `https://cdn.segment.com`
+const getGlobalCDNUrl = (): string | undefined => {
+  if (window.analytics) {
+    return window.analytics._cdn
   }
+}
 
-  return cdn
+export const setGlobalCDNUrl = (cdn: string) => {
+  if (window.analytics) {
+    window.analytics._cdn = cdn
+  }
+}
+
+export const getCDN = (): string => {
+  const globalCdnUrl = getGlobalCDNUrl()
+
+  if (globalCdnUrl) return globalCdnUrl
+
+  const cdnFromScriptTag = getCDNUrlFromScriptTag()
+
+  if (cdnFromScriptTag) {
+    setGlobalCDNUrl(cdnFromScriptTag)
+    return cdnFromScriptTag
+  } else {
+    // it's possible that the CDN is not found in the page because:
+    // - the script is loaded through a proxy
+    // - the script is removed after execution
+    // in this case, we fall back to the default Segment CDN
+    return `https://cdn.segment.com`
```

Comment:
Adding such a fallback means extra bytes of a string that can't be minified.

Reply:
Per the comment (which isn't mine, btw), we still need to have the fallback in case the script is removed after execution.

 If there's an easy way to keep this default constant in a way that you think is better for minification, I'm open to it. Otherwise, I would lean towards keeping the old behavior.